Source/Core/Runtime/SceneObjects/ProcessSceneReferenceBase.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using VRBuilder.Core.Runtime.Properties;
+
+namespace VRBuilder.Core.SceneObjects
+{
+    /// <summary>
+    /// Base class for a process reference to one or more objects.
+    /// </summary>
+    [DataContract(IsReference = true)]
+    public abstract class ProcessSceneReferenceBase : ICanBeEmpty
+    {
+        /// <summary>
+        /// List of guids, each Guid is a reference to a <see cref="Settings.SceneObjectTags.Tag"/>.
+        /// </summary>
+        public IEnumerable<Guid> Guids => guids;
```

Comment:
Changed to IReadOnlyList and switched from Count() to Count wherever possible. I'm not sure it makes much of a difference since that list is unlikely to contain more than 5 elements, but since it's accessed every frame it doesn't hurt.